functional-tests/src/test/groovy/fixtures/FixtureManager.groovy

```diff
@@ -0,0 +1,195 @@
+package fixtures
+
+import groovy.json.JsonSlurper
+
+import com.mongodb.*
+
+class FixtureManager {
+  private static String fixtureDataFolderPath = new File("src/test/groovy/fixtures/data").absolutePath;
+
+  private MongoClient client;
+  private DB db;
+
+  // Default connection values
+  private String host = 'localhost';
+  private int port = 27017;
+  private String databaseName = 'mem-dev-func';
+
+  // Stores fixture data that is inserted via this manager.
+  private ArrayList fixtureDataInserted = [];
+
+  /**
+   * A data fixture manager.
+   *
+   * This manager handles connecting to the mongo database and contains helper methods for inserting and
+   * removing data fixtures.
+   *
+   * When finished with this manager, it is recommneded that you call FixtureManager#removeAll() which will
+   * remove any fixture data that was added by this instance of the manager.
+   */
+  FixtureManager(final String host = null, final Integer port = null, final String databaseName = null) {
+    def env = System.getenv()
+    if(host) {
+      this.host = host;
+    } else if (env['MONGODB_FUNC_HOST']) {
+      this.host = env['MONGODB_FUNC_HOST'];
+    }
+
+    if (port) {
+      this.port = port;
+    } else if (env['MONGODB_FUNC_PORT']) {
+      this.port = env['MONGODB_FUNC_PORT'] as int;
+    }
+
+    if (databaseName) {
+      this.databaseName = databaseName;
+    } else if (env['MONGODB_FUNC_DATABASE'] ) {
+      this.databaseName = env['MONGODB_FUNC_DATABASE'];
+    }
+
+    this.client = new MongoClient(this.host, this.port);
+
+    this.db = client.getDB(this.databaseName);
+  }
+
+  /**
+   * @return the database instance used by this manager.
+   */
+  public getDB() {
+    return this.db;
+  }
+
+  /**
+   * Gets the fixture and returns the json object.
+   * @param fixture file name (required).
+   * @return an ArrayListMap of the fixture data or null if no data file is found.
+   */
+  public Object getFixture(String fixtureName) {
```

Comment:
Can we add the actual return type to the method signature, or Groovy doesn't support it?

Reply:
Changing to use List everywhere seems to work